### PR TITLE
Add Aug 30, 2021 meeting notes

### PR DESCRIPTION
**Pull request by acoburn**
This adds the minutes from the August 30, 2021, Solid Authentication Panel meeting.

### CODE REVIEW

File: meetings/2021-08-30.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# 2021-08-23 Solid Authentication
```
Comment (csarven) [inline meetings/2021-08-30.md:1]:
```suggestion
# 2021-08-30 Solid Authentication
```

**Review by elf-pavlik — APPROVED** [review]

File: meetings/2021-08-30.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Nicolas: when using an existing OIDC provider implementation, it was found that a `resource` parameter was supported
+as part of [RFC 8707 Resource Indicators for OAuth 2.0](https://datatracker.ietf.org/doc/html/rfc8707). This can be used
+to indicate which token to use with a resource server. This is relevant when a single provider is used with several
+different resource servers. The decentralized nature of Solid makes it hard to identify in advance which RS will be
+used for each OIDC provider. Perhaps one approach would be to indicate "any Solid resource" as a hint that the intention
+is to use the access token with Solid resources rather than some other, non-Solid resource.
+
+Pavlik: interesting. Have you looked at [draft PR 18](https://github.com/solid/solid-oidc/pull/18) where the IdP does
+not issue access tokens but just Id tokens. With that proposed architecture, there would be an association between
```
Comment (TallTed) [inline meetings/2021-08-30.md:32]:
```suggestion
Pavlik: interesting. Have you looked at [draft PR 18](https://github.com/solid/solid-oidc/pull/18), where the IdP does
not issue access tokens but just ID tokens? With that proposed architecture, there would be an association between
```

File: meetings/2021-08-30.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+an RS and an AS, but that would be independent of the Solid-OIDC spec requirements.
+
+Nicolas: aware of this direction.
+
+Pavlik: this suggestion might be part of a transitional state, but in the final spec, we may not want to have the
+IdP issue access tokens at all. Then, the resource server defines which AS to use. Clients would identify the
+authorization agent. In this structure, we wouldn't want a global access token issed by the IdP, as we have now.
+
+Aaron: I see two question here. First, is it in scope for Solid-OIDC and second, is it relevant for Solid?
+Given what Pavlik said that an IdP focuses only on identity, I would say that this is out-of-scope for Solid-OIDC.
```
Comment (TallTed) [inline meetings/2021-08-30.md:42]:
```suggestion
Aaron: I see two question here. First, is it in scope for Solid-OIDC; and second, is it relevant for Solid?
Given what Pavlik said — that an IdP focuses only on identity — I would say that this is out-of-scope for Solid-OIDC.
```

File: meetings/2021-08-30.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+Pavlik: there is an interop issue here. Having a multi-authorization-protocol ecosystem adds complexity.
+Do implementers need to support both protocols? That is a higher burden. By default, clients should not
+have access to resources; there should be a single path for authentication/authorization.
+These two topics need to be addressed together.
+
+Nicolas: will update issue to clarify that it is out of scope for Solid-OIDC but relevant for Solid.
+
+Pavlik: it's important to clearly define the scope of Solid-OIDC, e.g. issuing access tokens seems
+like it should be out of its scope.
```
Comment (TallTed) [inline meetings/2021-08-30.md:59]:
```suggestion
Pavlik: it's important to clearly define the scope of Solid-OIDC. For example, issuing access
tokens seems like it should be out of its scope.
```

File: meetings/2021-08-30.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+### Scope of Solid-OIDC
+
+Aaron: I wanted to ask Nic about this: how do client libraries handle Access Tokens vs Id Tokens.
+We have currently IdPs which issue both to the client. Per OIDC, the Id Token isn't usable to access RS.
+If we pursue the direction we just discussed, are there adequate protections for the Id Token, such that
+it could be exchanged for an Access Token. In particular, should we talk about DPoP binding for the Id Token?
+
+Nicolas: DPoP binding might apply to Id token as well as Access Token. In the current
+[solid-client-authn-js](https://github.com/inrupt/solid-client-authn-js) implementation, the Id Token is stored
+in local storage. It is only used when performing silent authentication (i.e. when a user refreshes a webpage).
```
Comment (TallTed) [inline meetings/2021-08-30.md:70]:
```suggestion
in local storage. It is only used when performing silent authentication (i.e., when a user refreshes a webpage).
```

File: meetings/2021-08-30.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+Nicolas: what would happen if a malicious user tampers with the issuer URL?
+
+Pavlik: because of the WebID trust model (solid:oidcIssuer), that sort of tampering would not give users any access.
+
+Nicolas: that sounds reasonable. I would need to think through the implications. Verifying the signature of the
+Id token may be unnecessary here.
+
+Pavlik: the RS provides the security here. We may need to clarify if there are other areas where malicious
+clients would tamper with data. It seems that an easy change would be for the client library to just store
+the Id token in a secure location (e.g. fetch closure).
```
Comment (TallTed) [inline meetings/2021-08-30.md:108]:
```suggestion
the Id token in a secure location (e.g., fetch closure).
```

File: meetings/2021-08-30.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+Nicolas: that _should_ be a simple change: managing the Id token inside that closure. We would need to update our documentation.
+
+Pavlik: DPoP may not even be needed with access tokens. We would definitely need to move the DPoP binding to Id tokens.
+
+Nicolas: DPoP binding seems like the mechanism we would need to use to prevent extfiltration attacks on Id tokens.
+
+Pavlik: we would need find consensus on this model: namely, that Solid-OIDC focuses only on Identity.
```
Comment (TallTed) [inline meetings/2021-08-30.md:116]:
```suggestion
Pavlik: we would need to find consensus on this model: namely, that Solid-OIDC focuses only on Identity.
```

File: meetings/2021-08-30.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+Pavlik: we should discuss that. That is the case in the interop panel. The Authorization panel is focused more on
+ACP/WAC at present. It is important the the requesting party can authorize clients. It seems that the interop panel
+has been incorporating the model of claims pushing. This should be added to an agenda in an upcoming meeting.
+
+Aaron: we should be careful about not breaking existing implementations and deployments all at once.
+There are both technical and community considerations.
+
+Barath: people will be upset if we just break things. We are running into it in context of Authorization.
+We were discussing weather they may be some overarching best practice for Solid security.
```
Comment (TallTed) [inline meetings/2021-08-30.md:128]:
```suggestion
We were discussing whether there may be some overarching best practice for Solid security.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
typos and such

**Pull request merged**